tapdb/multiverse_cache.go

```diff
@@ -187,6 +189,212 @@ func (r *rootIndex) wipe() {
 	r.Store(&idx)
 }
 
+// syncerRootNodeCache is used to cache the set of active root nodes for the
+// multiverse tree, which is specifically kept for the universe sync.
+type syncerRootNodeCache struct {
+	sync.RWMutex
+
+	// universeKeyList is the list of all keys of the universes that are
+	// currently known in this multiverse. This slice is sorted by the key
+	// (which might differ from the database ordering) and can be used to
+	// paginate through the universes efficiently. The universe identifier
+	// key also contains the proof type, so there will be two entries
+	// per asset ID or group key, one for issuance and one for transfer
+	// universes. The universe roots map below will contain a root for each
+	// of these universes.
+	universeKeyList []universe.IdentifierKey
+
+	// universeRoots is a map of universe ID key to the root of the
+	// universe. This map is needed to look up the root of an individual
+	// universe quickly, or to look up the roots of all the universes when
+	// paging through them. This map never needs to be cleared, as universe
+	// roots are only added (new issuance or transfer events) or modified
+	// (re-issuance into grouped asset or new transfer) but never removed.
+	// Therefore, when the cache needs to be wiped, we only need to clear
```

Comment:
>  but never removed.

It's not exposed over the RPC interface rn, but they _can_ be removed (see the `DeleteUniverse` method).